updated frontend

Quillraven opened the pull request:
This feature is based on #10 .

Added routing to frontend. We now have an overview page and a separate details page. In addition, we have a Navigationbar at the top, which does not do anything yet, but it could potentially open a drawer and the account settings page.

Clicking on a toilet in the overview brings you to the details page.

Clicking on the back button on the details page brings you back to the overview.

----

![image](https://user-images.githubusercontent.com/93260/96386627-ab36d580-119c-11eb-92fa-15920c5d914e.png)

![image](https://user-images.githubusercontent.com/93260/96386654-d3becf80-119c-11eb-8acb-28fd944c92b7.png)

![image](https://user-images.githubusercontent.com/93260/96386641-be49a580-119c-11eb-80f5-73c87e13130c.png)

christian-baumann commented:
An "high risk vulnerability" was introduces (npm install output), which should no be done at least with new code. 

UI remarks: 

* Toilet-Details view: Rating widget should be bigger (there are sized XS, S, M, L)
* Toilet-Details view: Probably the image should not be display in full width (at least for big screens)
* Toilet-List view: Toilet list should be a responsive grid the be able to utilize the full screen no matter which size

Nevertheless, the UI looks good and can be merged as soon as the vulnerability is fixed and we can create new issues for the improvements.

Quillraven commented:
@christian-baumann can you try again please? I fixed our versions now to apply with the official versions of spring for the backend, and I used the latest versions for our frontend.
I removed `node-forge` ... I cannot remember why I added it but it does not seem to be necessary.

Also, latest material-ui renamed `ExpansionPanel` to `Accordion`. I changed that as well.

For me it looks fine and I don't got vulnerabilities. Please check it on your side and if everything is fine, I will merge it.
